mq/mq_test.go

```diff
@@ -380,6 +385,34 @@ var _ = Describe("MQ", func() {
 		})
 	})
 
+	Context("when dropping all messages below a certain height", func() {
+		It("should remove all corresponding messages from the queues", func() {
+			opts := mq.DefaultOptions()
+			queue := mq.New(opts)
+
+			loop := func() bool {
+				sender := id.NewPrivKey().Signatory()
+				_, maxHeight, _ := insertRandomMessages(&queue, sender)
+				thresholdHeight := process.Height(r.Intn(int(maxHeight)))
+				queue.DropMessagesBelowHeight(thresholdHeight)
+
+				proposeCallback := func(propose process.Propose) {
+					Expect(propose.Height >= thresholdHeight).To(BeTrue())
+				}
+				prevoteCallback := func(prevote process.Prevote) {
+					Expect(prevote.Height >= thresholdHeight).To(BeTrue())
+				}
+				precommitCallback := func(precommit process.Precommit) {
+					Expect(precommit.Height >= thresholdHeight).To(BeTrue())
+				}
+
+				_ = queue.Consume(maxHeight, proposeCallback, prevoteCallback, precommitCallback)
```

Comment:
Should we be checking the value returned by consume?

Reply:
We could, though it would require computing how many messages we expect to get dropped which is kind of annoying. I decided not to because it wasn't the main focus of the test.

Reply:
Oh, sorry I misunderstood what the test was doing. Makes sense.